src/modules/analytics/services/global.rewards.service.ts

```diff
@@ -0,0 +1,266 @@
+    })
+    async farmRewards(weekOffset: number): Promise<FarmsGlobalRewards[]> {
+        try {
+            return await this.processFarmsRewards(weekOffset);
+        } catch (error) {
+            this.logger.error(`Error getting farms rewards: ${error.message}`);
+            throw new Error('Error getting farms rewards');
+        }
+    }
+
+    @GetOrSetCache({
+        baseKey: 'analytics',
+        remoteTtl: Constants.oneHour() * 6,
+        localTtl: Constants.oneHour() * 4,
+    })
+    async stakingRewards(weekOffset: number): Promise<StakingGlobalRewards[]> {
+        try {
+            return await this.processStakingRewards(weekOffset);
+        } catch (error) {
+            this.logger.error(
+                `Error getting staking rewards: ${error.message}`,
+            );
+            throw new Error('Error getting staking rewards');
+        }
+    }
+
+    private async processFeesCollectorRewards(
+        weekOffset: number,
+    ): Promise<FeesCollectorGlobalRewards> {
+        let energyRewardsUSD = new BigNumber(0);
+        const feesCollectorAddress = scAddress.feesCollector;
+
+        const currentWeek =
+            await this.weeklyRewardsSplitting.lastGlobalUpdateWeek(
+                feesCollectorAddress,
+            );
+
+        const targetWeek = Math.max(0, currentWeek - weekOffset);
+
+        const weekRewardsUSD =
+            await this.weeklyRewardsSplittingCompute.totalRewardsForWeekUSD(
+                feesCollectorAddress,
+                targetWeek,
+            );
+
+        if (weekRewardsUSD && weekRewardsUSD !== '0') {
+            energyRewardsUSD = energyRewardsUSD.plus(
+                new BigNumber(weekRewardsUSD),
+            );
+        }
+
+        const totalRewardsUSD = energyRewardsUSD.dividedBy(0.6);
+
+        return new FeesCollectorGlobalRewards({
+            totalRewardsUSD: totalRewardsUSD.toFixed(),
+            energyRewardsUSD: energyRewardsUSD.toFixed(),
+        });
+    }
+
+    private async processFarmsRewards(
+        weekOffset: number,
+    ): Promise<FarmsGlobalRewards[]> {
+        const farmsData: FarmsGlobalRewards[] = [];
+        const farmAddresses = farmsAddresses([FarmVersion.V2]);
+
+        const pairAddresses = await this.farmAbiV2.getAllPairContractAddresses(
+            farmAddresses,
+        );
+
+        const firstTokens = await this.pairService.getAllFirstTokens(
+            pairAddresses,
+        );
+        const secondTokens = await this.pairService.getAllSecondTokens(
+            pairAddresses,
+        );
+        const allPairsTokens = [...firstTokens, ...secondTokens];
+
+        for (let i = 0; i < farmAddresses.length; i++) {
+            const farmAddress = farmAddresses[i];
+
+            const currentWeek = await this.weekTimekeepingAbi.currentWeek(
+                farmAddress,
+            );
+            const weekToCheck = Math.max(0, currentWeek - weekOffset);
+
+            const rewards =
```

Comment:
You have to use other method for currentWeek: `farmAbi.accumulatedRewardsForWeek`. Check if `weekOffset === 0`